app/models/subnet/ipv4.rb

```diff
@@ -0,0 +1,38 @@
+require 'socket'
+
+class Subnet::Ipv4 < Subnet
+  has_many :interfaces, :class_name => 'Nic::Base', :foreign_key => :subnet_id
+  has_many :primary_interfaces, -> { where(:primary => true) }, :class_name => 'Nic::Base', :foreign_key => :subnet_id
+  has_many :hosts, :through => :interfaces
+  has_many :primary_hosts, :through => :primary_interfaces, :source => :host
```

Comment:
Why moving these last 2 has_many? I can see why the other 2 need to be moved, because `:foreign_key` can also be `:subnet6_id`, but these two don't have that limitation